src/build_vehicle_gui.cpp

```diff
@@ -1335,6 +1364,27 @@ struct BuildVehicleWindow : Window {
 		return CargoAndEngineFilter(&item, filter_type);
 	}
 
+	/** Filter by name and NewGRF extra text */
+	bool FilterByText(const Engine *e)
+	{
+		/* Do not filter if the filter text box is empty */
+		if (this->string_filter.IsEmpty()) {
+			return true;
+		}
+
+		/* Filter engine name */
+		char buffer[DRAW_STRING_BUFFER];
+		GetString(buffer, e->info.string_id, lastof(buffer));
```

Comment:
Since you added a `std::string` `AddLine` version, you can use the `std::string` returning `GetString` here.